jumpscale/sals/marketplace/apps_chatflow.py

```diff
@@ -334,7 +338,20 @@ def get_solution_name(self):
     def infrastructure_setup(self):
         self.md_show_update("Preparing Infrastructure...")
         self._get_pool()
-        self._deploy_network()
+        success = False
+        while not success:
+            try:
+                self._deploy_network()
+                success = True
+            except DeploymentFailed as e:
+                j.logger.error(e)
+                if self.retries > 0:
+                    self.md_show_update(f"Deployment failed on node {self.selected_node.node_id}. retrying....")
+                    self.retries -= 1
+                    self.ip_address = None
+                    self._deploy_network()
+                else:
+                    raise e
```

Comment:
```suggestion
                    raise StopChatFlow(f"Failed to deploy this instance: { str(e) }")
```

Reply:
`e` is an instance of `DeploymentFailed`. I need to raise it exactly as it is to properly clean the workloads